lib/srv/db/sqlserver/engine.go

```diff
@@ -162,6 +179,27 @@ func (e *Engine) receiveFromClient(clientConn, serverConn io.ReadWriteCloser, cl
 	}
 }
 
+// toSQLPacket Parses a regular (self-contained) or chunked packet into an SQL
+// packet (used for auditing).
+func (e *Engine) toSQLPacket(header protocol.PacketHeader, packet *protocol.BasicPacket, chunks *bytes.Buffer) (protocol.Packet, error) {
+	if chunks.Len() > 0 {
+		defer chunks.Reset()
+		chunks.Write(packet.Data())
+		packetData, err := io.ReadAll(chunks)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
```

Comment:
I think we can do `chunks.Bytes()` to get all the bytes (given that we don't do partial readings)

Reply:
Yeah, we can do that, actually. Initially, I thought we weren't copying the data from the chunks (which could lead to issues as the slice returned by `Bytes()` is only valid until the next non-read operation, in our case, `Reset()`). But the contents used by audit events are copied for both queries and RPCs. Otherwise, our test would fail because we assert events contents after everything is consumed.

I'll update the code and add a comment. Thanks!